### PR TITLE
[feat] Notes BE implementation

### PR DESCRIPTION
**Pull request by devfox-se**

### CODE REVIEW

**Comment by devfox-se** [comment]
Implements #1607

File: aim/storage/structured/sql_engine/models.py [hunk 1]
```diff
@@ -85,3 +93,45 @@ def __init__(self, name):
...
+
+
+class Note(Base):
+    __tablename__ = 'note'
+    __table_args__ = (
+        UniqueConstraint('run_id', 'name'),
+    )
+
+    id = Column(Integer, autoincrement=True, primary_key=True)
+    name = Column(Text, nullable=False, default='New Note')
```
Comment (alberttorosyan) [inline aim/storage/structured/sql_engine/models.py:105]: what the `name` field is used for?

File: aim/storage/structured/sql_engine/models.py [hunk 2]
```diff
@@ -85,3 +93,45 @@ def __init__(self, name):
     def validate_color(self, _, color):
         # TODO: [AT] add color validation
         return color
+
+
+class Note(Base):
+    __tablename__ = 'note'
+    __table_args__ = (
```
Comment (alberttorosyan) [inline aim/storage/structured/sql_engine/models.py:100]: do we need this constraint? why restrict someone to add the same note (comment) multiple times?

File: aim/web/api/runs/note/pydnamic_models.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from pydantic import BaseModel
```
Comment (alberttorosyan) [inline aim/web/api/runs/note/pydnamic_models.py:1]: please move the APIs to `runs/views.py`
the directory structure becomes too nested here.
same for the inout models

File: aim/storage/structured/sql_engine/entities.py [hunk 4]
```diff
@@ -175,6 +180,78 @@ def remove_tag(self, tag_name: str) -> bool:
         session.flush()
         return tag_removed
 
+    @property
+    def notes(self) -> List[dict]:
```
Comment (alberttorosyan) [inline aim/storage/structured/sql_engine/entities.py:184]: worth adding `notes_obj` method like it is done for tags 🤔

**Review by alberttorosyan — CHANGES_REQUESTED** [review]
left some comments.
please add the unit-tests as well

File: aim/storage/structured/sql_engine/models.py (hunk 1 above)
Comment (devfox-se) [reply aim/storage/structured/sql_engine/models.py:105]: @alberttorosyan treat is as a file name where you write your notes

File: aim/storage/structured/sql_engine/entities.py (hunk 4 above)
Comment (devfox-se) [reply aim/storage/structured/sql_engine/entities.py:184]: maybe... but what would it give us?

File: aim/storage/structured/sql_engine/models.py (hunk 2 above)
Comment (devfox-se) [reply aim/storage/structured/sql_engine/models.py:100]: @alberttorosyan its like having `new folder` and `new folder` on your desktop :D how would you distinguish ?

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Pull request merged**